### PR TITLE
Add Data Maturity Assessment to website

### PR DESCRIPTION
**Pull request by dpblume**
Adding new page & survey to host our freebie data maturity assessment

Current screenshot:
image

TODO: 
1. Add backend to actually store the response data
2. Add content on the landing page to actually direct people to this page & make them aware of it

### CODE REVIEW

File: assets/js/survey.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  // POST /submit-survey
+  fetch('/submit-survey', {
+      method: 'POST',
+      headers: {
+          'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(formData)
+  })
+  .then(response => response.json())
+  .then(data => {
```
Comment (nathanaelrobinson) [inline assets/js/survey.js:64]: we can update this probably want to make sure that the response.code is in the 400 range

**Comment by dpblume** [comment]
Love that, great idea! I'll take a stab at it

**Comment by dpblume** [comment]
Updates: users now step through question by question

Remain items:
- I've tried a bunch of different things but I can't seem to make the previous/next buttons side-by-side. They keep stacking on top of each other
- Need to add in a bail-out "unsure" question to each answer that doesn't currently have one
- No backend steps have been set up yet to store results

image

**Comment by nathanaelrobinson** [comment]
> Updates: users now step through question by question
> 
> Remain items:
> 
> * I've tried a bunch of different things but I can't seem to make the previous/next buttons side-by-side. They keep stacking on top of each other
> * Need to add in a bail-out "unsure" question to each answer that doesn't currently have one
> * No backend steps have been set up yet to store results
> 
> image

for #2 this is a css issue. You'll want to have a single div that encompasses both buttons -- and then set the style on that div to flex and then then the spacing as either space-between or space-around.

I'll provide an example - this is funny this is literally the hardest html/css concept for my students when I teach my web dev class so I've gone through this problem a billion times

File: templates/data_assessment.html [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+        margin: 0 auto;
+        padding: 20px;
+        background-color: #f9f9f9;
+        border-radius: 8px;
+        box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.1);
+    }
+
+    .submit-container {
+        max-width: 60%;
+    }
```
Comment (nathanaelrobinson) [inline templates/data_assessment.html:56]:
```suggestion
    .submit-container {
        max-width: 60%;
        display: flex;
        justify-content: space-between;
    }
```

See the flexbox documentation
https://css-tricks.com/snippets/css/a-guide-to-flexbox/

**Comment by nathanaelrobinson** [comment]
Also RE: Backend -- we will need to issue a migration to the database and great an ORM object for you to interact with -- so let me to ahead and do that in the background and get that merged to the Cronos app first.

File: templates/data_assessment.html (hunk 2 above)
Comment (dpblume) [reply templates/data_assessment.html:56]: This worked great, thanks!

File: templates/data_assessment.html [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+                <input type="radio" id="q15d" name="q15" value="no-defined-metrics">
+                <label class="answer-option-label" for="q15d">We don’t currently have any defined metrics or KPIs</label>
+            </div>
+            <textarea name="q15_text" placeholder="Optional: Provide additional details"></textarea>
+            <div class="error-message" style="display:none;"></div>
+        </div>
+
+        <!-- Navigation Buttons -->
+        <div class="submit-container">
+            <button type="button" id="prev-btn" style="display:none;">Previous</button>
```
Comment (nathanaelrobinson) [inline templates/data_assessment.html:441]:
```suggestion
           <button type="button" id="start-btn">Start</button>
```

Create a unique start button to begin that way you can push it to the right side
you've got two options with CSS -- I'd suggest the first:
1. `flex-direction: row-reverse;`
2. `float: right;`

File: main.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ func main() {
 		// which must be established via a local SQLite instance, you will need to
 		// run the migration to create the database schema
 		cronosApp.InitializeSQLite()
-		//cronosApp.Migrate()
+		cronosApp.Migrate()
```
Comment (dpblume) [inline main.go:54]: @nathanaelrobinson does this need to be commented back out before final review / merge?

Comment (nathanaelrobinson) [reply main.go:54]: It only triggers in the else loop when we already initialize a sqlite instance so it's fine to keep, in fact we probably should keep it so people don't get stale local dbs for testing.

**Review by nathanaelrobinson — APPROVED** [review]
I've got no issues, runs smoothly on my machine and code looks great let's 🚢  this guy!

**Pull request merged**